tests/integration/vlan_test.py

```diff
@@ -90,9 +91,39 @@ def test_add_and_remove_two_vlans_on_same_iface(eth1_up):
     assert not current_state[Interface.KEY]
 
 
+def test_two_vlans_on_eth1_change_mtu(eth1_up):
+    desired_state = statelib.show_only(('eth1',))
+    vlans_state = copy.deepcopy(TWO_VLANS_STATE)
+    desired_state[Interface.KEY].extend(vlans_state[Interface.KEY])
+    for iface in desired_state[Interface.KEY]:
+        iface[Interface.MTU] = 2000
+    libnmstate.apply(desired_state)
+
+    for iface in desired_state[Interface.KEY]:
+        if iface[Interface.NAME] != 'eth1.101':
```

Comment:
You can save this base_iface_state in advance with no trouble to avoid this strange lookup.

Reply:
I do not understand this. In this look up I am setting the MTU of all interfaces from the desired state except the "eth1.101" interface.

Reply:
You know in the test who are these "all" except "eth1.101": "eth1" and "eth1.102".
In general, using `for` loops in tests is confusing, it requires interpreting what has been done instead of being explicit. In this case, I got confused and thought you want to change just "eth1" (the base).
I prefer to see what you changed explicitly.

Could you also use the names constants?

Reply:
Done! Thanks for explaining